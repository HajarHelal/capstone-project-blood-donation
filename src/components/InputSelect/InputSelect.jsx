import React from 'react';
import { PropTypes } from 'prop-types';

function InputSelect({ donor, text, onChange, name, options }) {
  return (
    <div className="mb-4 sm:mb-6">
      <label htmlFor="nowhere" className="block">
        <div className="text-sm sm:text-base md:text-lg font-medium inline-block mb-4 sm:mb-6">
          {text}
        </div>
        <select
          className="form-select block w-full border-gray-300 border-2 py-2 px-4 text-sm sm:text-base md:text-lg "
          value={donor[name]}
          name={name}
          onChange={onChange}
          onBlur={onChange}
        >
          {options.map((ele) => (
            <option key={ele}>{ele}</option>
          ))}
        </select>
      </label>
    </div>
  );
}

InputSelect.propTypes = {
  donor: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
};
export default InputSelect;
