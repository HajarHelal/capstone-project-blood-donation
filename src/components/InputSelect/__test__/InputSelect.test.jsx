import React from 'react';
import renderer from 'react-test-renderer';
import InputSelect from '../InputSelect';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(
      <InputSelect
        city="Abyan"
        onChange={() => console.log('donate')}
        name="city"
        text="3. Please select your city ?*"
        options={[
          'Abyan',
          'Aden',
          'Al Mahrah',
          'Hadramaut',
          'Lahij',
          'Shabwah',
          'Dhale',
          'Amran',
          'Al Bayda',
          'Al Hudaydah',
          'Al Jawf',
          'Al Mahwit',
          'Dhamar',
          'Hajjah',
          'Ibb',
          "Ma'rib",
          'Raymah',
          'Saada',
          "Sana'a",
          'Taiz',
          'Socotra',
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
