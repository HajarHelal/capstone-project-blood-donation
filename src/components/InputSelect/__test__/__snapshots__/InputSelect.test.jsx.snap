// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mb-4 sm:mb-6"
>
  <label
    className="block"
    htmlFor="nowhere"
  >
    <div
      className="font-medium text-sm sm:text-base md:text-lg  inline-block mb-4 sm:mb-6"
    >
      3. Please select your city ?*
    </div>
    <select
      className="form-select block w-full border-gray-300 border-2 py-2 px-4 text-sm sm:text-base md:text-lg "
      name="city"
      onBlur={[Function]}
      onChange={[Function]}
      value="Abyan"
    >
      <option>
        Abyan
      </option>
      <option>
        Aden
      </option>
      <option>
        Al Mahrah
      </option>
      <option>
        Hadramaut
      </option>
      <option>
        Lahij
      </option>
      <option>
        Shabwah
      </option>
      <option>
        Dhale
      </option>
      <option>
        Amran
      </option>
      <option>
        Al Bayda
      </option>
      <option>
        Al Hudaydah
      </option>
      <option>
        Al Jawf
      </option>
      <option>
        Al Mahwit
      </option>
      <option>
        Dhamar
      </option>
      <option>
        Hajjah
      </option>
      <option>
        Ibb
      </option>
      <option>
        Ma'rib
      </option>
      <option>
        Raymah
      </option>
      <option>
        Saada
      </option>
      <option>
        Sana'a
      </option>
      <option>
        Taiz
      </option>
      <option>
        Socotra
      </option>
    </select>
  </label>
</div>
`;
